### DISCUSSION TITLE
What language/platform-specific examples/templates would you like to see next?

### DISCUSSION THREAD

**Discussion by jpadams** [dagger/dagger#2874] [General]
We have a repo to hold Dagger examples including what we're calling "templates": small, ready-to-run examples of Dagger being used for a specific platform or language (think Node.js, Java, Python, etc). The templates are currently mostly "build" examples, but that could be expanded to simple "deploy" or "test" scenarios as well.

https://github.com/dagger/examples

If there is an example template that you'd like to see, please create a comment below. Ideally, one comment per template type so folks can upvote ones they like. Or feel free to vote with your keyboard and make a PR :)

Thanks!

**Comment by jritsema** [comment]
This URL 404s for me - https://github.com/dagger/examples

**Reply by jpadams** [reply]
Huh...still 404? I clicked your link
<img width="200" alt="image" src="https://user-images.githubusercontent.com/3187222/183738699-7a2230f3-5b3b-42fc-8215-3a663b2c63a2.png">

**Reply by TomChv** [reply]
It's a private repository right now, that's why

**Reply by jpadams** [reply]
Wow! Thanks, @TomChv ! Right in front of me 🤣 Thanks, @jritsema !

**Reply by jritsema** [reply]
Works now!  Thanks guys.

**Comment by jritsema** [comment]
I would like to see some examples of deploying a node app to ECS Fargate.

**Reply by jpadams** [reply]
@jritsema do you have an example in the AWS docs/examples that would be a good first example to emulate?

Guessing it's:
- copy in source code
- build an image
- push to ECR
- then the harder parts :) ECS task definition in `aws` cli or CFT or Terraform or Pulumi or ...

**Comment by nipuna-perera** [comment]
I would love to see some terraform examples. Like a lambda function
deployment.

**Reply by aavileli** [reply]
Templates
- building a maven project with .m2 cache
- pushing to ecr
- cloning another repository and executing helm manifest and commiting to git after pushing to ecr 
- running terraform and waiting for user approval
- using buildkit on kubernetes

**Comment by FlxPeters** [comment]
Ansible deployment to a Virtual Machine or network devices would be cool including ansible-lint and molecule role tests.

**Comment by albe2669** [comment]
Docker build and push to container repository would be cool to have as an example

**Comment by tomiberes** [comment]
A "go-sdk" equivalent of `docker build -t service/name .` and `docker run -d service/name`.

**Reply by jpadams** [reply]
Support for running services like `docker run -d` is still under development. Stay tuned!

The `docker build` from a `Dockerfile` example works and is in the reference docs today here: https://pkg.go.dev/dagger.io/dagger#Container.Build:~:text=a%20Dockerfile%20build-,Example
Use `docker login` in your terminal before running to push to a registry that needs auth.

That example shows pulling the filesystem with the `Dockerfile` from a git repo, but you could read from the host filesystem as well like this:

My `main.go`:
```go
package main

import (
	"context"
	"fmt"
	"os"

	"dagger.io/dagger"
)

func main() {
	ctx := context.Background()

	client, err := dagger.Connect(ctx, dagger.WithLogOutput(os.Stdout))
	if err != nil {
		fmt.Printf("Error connecting to Dagger Engine: %s", err)
		os.Exit(1)
	}

	defer client.Close()

	src := client.Host().Workdir()
	if err != nil {
		fmt.Printf("Error getting reference to host directory: %s", err)
		os.Exit(1)
	}
	_, err = client.Container().
		Build(src).
		Publish(ctx, "jeremyatdockerhub/gosdk-example:latest")

	if err != nil {
		fmt.Printf("Error creating and pushing container: %s", err)
		os.Exit(1)
	}
}
```

`Dockerfile` I used:
```
FROM alpine
CMD ["echo", "Hello World"]
```
